src/transaction/log_recovery_redo_parallel.cpp

```diff
@@ -779,75 +781,119 @@ namespace cublog
    *********************************************************************/
 
   reusable_jobs_stack::reusable_jobs_stack ()
-    : m_stack_size { 0 }
+    : m_job_count { 0 }
+    , m_push_task_count { 0 }
+    , m_flush_push_at_count { 0 }
+    , m_jobs_arr { nullptr }
   {
   }
 
-  void reusable_jobs_stack::initialize (std::size_t a_stack_size, const log_lsa *a_end_redo_lsa,
-					bool force_each_page_fetch)
+  void reusable_jobs_stack::initialize (std::size_t a_job_count, std::size_t a_push_task_count,
+					std::size_t a_flush_push_at_count,
+					const log_lsa *a_end_redo_lsa, bool force_each_page_fetch)
   {
-    assert (m_pop_stack.empty ());
-    assert (m_push_stack.empty ());
+    assert (m_job_count == 0);
+    assert (m_push_task_count == 0);
+    assert (m_flush_push_at_count == 0);
 
-    m_stack_size = a_stack_size;
-    for (std::size_t idx = 0; idx < m_stack_size; ++idx)
+    assert (m_jobs_arr == nullptr);
+
+    assert (m_pop_jobs.empty ());
+    assert (m_push_jobs.empty ());
+
+    assert (m_per_task_push_jobs_vec.empty ());
+
+    assert (a_job_count > 0);
+    assert (a_push_task_count > 0);
+    assert (a_flush_push_at_count > 0);
+
+    m_job_count = a_job_count;
+    m_push_task_count = a_push_task_count;
+    m_flush_push_at_count = a_flush_push_at_count;
+
+    m_jobs_arr = static_cast<unsigned char *> (malloc (sizeof (redo_job_impl) * m_job_count));
```

Comment:
```suggestion
    m_jobs_arr = new redo_job_impl[m_job_count];
```

Reply:
`redo_job_impl` does not have a default ctor; via its ctor a part of the params are initialized; and then, at each reuse, another part of the params are initialized via `reinitialize`; that is the reason of the malloc + placement new + free